src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -351,6 +355,33 @@ class Config:
         return await self.middleware.call('datastore.delete', self._config.datastore, id)
 
 
+def _run_command(user, commandline, q, rv):
+    setusercontext(user)
```

Comment:
Why not `su`? In which case we won't need `multiprocessing` here at all.

Reply:
I am simply moving code around, not reinventing the wheel.

IIRC su did not set all user context because it has some limitations regarding users and their shell.
We can eventually investigate it further.

Reply:
The original code is in tools/runow.py